src/config/app-config-endpoints.html

```diff
@@ -6,6 +6,39 @@
     domain: '.domain.com'
   };
 
+  function polyfillLocationOrigin() {
+    if (window.location.origin) { return; }
+
+    window.location.origin = window.location.protocol + "//" + window.location.hostname + (window.location.port ? ":" + window.location.port : "");
+  }
+
+  function resolveApiBaseUrl() {
+    const config = AppEndpoints;
+    const hostname = window.location.hostname;
+    let apiBaseUrl;
+
+    if (hostname.indexOf('.') === -1) {
+      apiBaseUrl = `${config.protocol}${config.subdomain}${config.domain}`;
+    } else {
+      const subdomain = window.location.hostname.split('.').shift();
+      let apiSubdomain;
+      if (!subdomain.endsWith('-api')) {
+        apiSubdomain = subdomain.concat('-api');
+      }
+
+      if (hostname.endsWith('localhost')) {
+        apiBaseUrl = `${config.protocol}${apiSubdomain}${config.domain}`;
```

Comment:
This is in my amended code, but it looks like you're missing a `.` between your last two variables.